core/templates/dev/head/domain/classifier/PredictionResultObjectFactory.js

```diff
@@ -0,0 +1,39 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of Prediction
+ *     result domain objects.
+ */
+
+oppia.factory('PredictionResultOjectFactory', [function() {
+  var predictionResult = function(label, confidence) {
```

Comment:
Would probably be a good idea to add a comment with some explanation of what the arg types + descriptions are (e.g. making clear whether "confidence" is a number between 0 and 1, or a percentage, etc.)